templates/Makefile.erb

```diff
@@ -79,8 +79,11 @@ cleanup-components: <%= @components.map {|comp| "#{comp.name}-cleanup" }.join("
 	touch <%= comp.name %>-patch
 
 <%= comp.name %>-configure: <%= comp.name %>-patch <%= list_component_dependencies(comp).join(" ") %>
+	<%- if comp.get_build_dir -%>
+	[ -d <%= comp.get_build_dir %> ] || mkdir <%= comp.get_build_dir %>
```

Comment:
It looks like mkdir will actually fail if the directory exists: 

```
└> mkdir foo; mkdir foo; echo $?
mkdir: cannot create directory ‘foo’: File exists
1
```

+1 for `mkdir -p`; adding it now.